e2e/test/helpers/templates/FaultInjection.cs

```diff
@@ -91,7 +91,7 @@ public static async Task ActivateFaultInjectionAsync(IotHubClientTransportSettin
 
                 await deviceClient.SendEventAsync(faultInjectionMessage, cts.Token).ConfigureAwait(false);
             }
-            catch (Exception ex) when (ex is IotHubCommunicationException || ex is TimeoutException)
+            catch (Exception ex) when (ex is IotHubClientException && ((IotHubClientException)ex).StatusCode is IotHubStatusCode.NetworkErrors || ex is TimeoutException)
```

Comment:
Where does `TimeoutException` come from? Can that be mapped to something else earlier?

Reply:
I removed `TimeoutException` here and ran locally, nothing was changed in particular.

Reply:
I will find a way to not throw TimeoutException in another exception-related PR.